kge/dataset.py

```diff
@@ -91,6 +93,53 @@ def load(config: Config, preload_data=True):
                 dataset.split(split)
         return dataset
 
+    @staticmethod
+    def create_from(
+        checkpoint: Dict,
+        config: Config,
+        dataset: Optional[Dataset] = None,
+        preload_data=True,
+    ) -> Dataset:
+        """
+        Creates dataset based on a checkpoint.
+        If a dataset is provided it will be updated
+        Args:
+            checkpoint: loaded checkpoint
+            config: config created from checkpoint
+            dataset: dataset to update
+            preload_data: preload data
+
+        Returns: created/updated dataset
+
+        """
+        if dataset is None:
+            dataset = Dataset.load(config, preload_data)
+        if "dataset" in checkpoint:
+            dataset_checkpoint = checkpoint["dataset"]
+            if (
+                "dataset.meta" in dataset_checkpoint
+                and dataset_checkpoint["meta"] is not None
+            ):
+                dataset._meta.update(dataset_checkpoint["meta"])
+            dataset._num_entities = dataset_checkpoint["num_entities"]
+            dataset._num_relations = dataset_checkpoint["num_relations"]
+        return dataset
+
+    def save_to(self, checkpoint: Dict, meta_keys: Optional[List[str]] = None) -> Dict:
+        """Adds specified meta data to a checkpoint"""
+        dataset_checkpoint = {
+            "num_entities": self.num_entities(),
+            "num_relations": self.num_relations(),
+        }
+        checkpoint["dataset"] = dataset_checkpoint
+        if meta_keys is None:
+            return checkpoint
+        meta_checkpoint = {}
+        for key in meta_keys:
+            meta_checkpoint[key] = self._map_indexes(None, key)
```

Comment:
_map_indexes seems to be incorrect here. Do you mean map_indexes? Also, why not just store meta_["key"]?

Reply:
I switched to map_indexes. If we just store meta["key"], we can not make sure, that we actually loaded this data. Could be that we didn't even read the entitiy_ids file, when calling this method.